dan_layer/state_store_sqlite/src/substate.rs

```diff
@@ -0,0 +1,10 @@
+// Copyright 2024 The Tari Project
+// SPDX-License-Identifier: BSD-3-Clause
+
+use tari_dan_storage::StorageError;
+use tari_engine_types::substate::SubstateId;
+
+/// General trait for different state stores to do operations on substates.
+pub trait SubstateStore {
```

Comment:
This doesnt seem right. Firstly, this is in the SQLite implementation of the state store which will be a problem when we switch to rocksDB. Templates can always be version 0, so not sure this is necessary, I'll need to look more closely at the rest of the code. There is currently no special process for upgrading the template code, you simply publish a new template at v0.